File: src/parser/python/utils/defvars.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

import platform
from pathlib import Path

# Builtin variables.
def builtins(cmdname):
    return {
        "HOME": str(Path.home()),
        "OS": platform.system().lower(),
        "COMMAND": cmdname,
        "PATH": f"~/.nodecliac/registry/{cmdname}"
    }

File: src/parser/nim2/defvars.nim
import std/[os, tables]

import utils/[osutils]

# Builtin variables.
proc builtins*(cmdname: string): Table[string, string] =
    return {
        "HOME": os.getEnv("HOME"),
        "OS": platform(),
        "COMMAND": cmdname,
        "PATH": "~/.nodecliac/registry/" & cmdname
    }.toTable
